vms/evm/sync/customrawdb/sync_progress.go

```diff
@@ -0,0 +1,235 @@
+}
+
+// ClearAllSyncSegments removes all segment markers from db
+func ClearAllSyncSegments(db ethdb.KeyValueStore) error {
+	return clearPrefix(db, syncSegmentsPrefix, syncSegmentsKeyLength)
+}
+
+// ParseSyncSegmentKey returns the root and start position for a trie segment
+// key returned from NewSyncSegmentsIterator.
+func ParseSyncSegmentKey(keyBytes []byte) (common.Hash, []byte) {
+	keyBytes = keyBytes[len(syncSegmentsPrefix):] // skip prefix
+	root := common.BytesToHash(keyBytes[:common.HashLength])
+	start := keyBytes[common.HashLength:]
+	return root, start
+}
+
+// NewSyncStorageTriesIterator returns a KeyLength iterator over all storage tries
+// added for syncing (beginning at seek). It is the caller's responsibility to parse
+// the key and call Release on the returned iterator.
+func NewSyncStorageTriesIterator(db ethdb.Iteratee, seek []byte) ethdb.Iterator {
+	return rawdb.NewKeyLengthIterator(db.NewIterator(syncStorageTriesPrefix, seek), syncStorageTriesKeyLength)
+}
+
+// WriteSyncStorageTrie adds a storage trie for account (with the given root) to be synced.
+func WriteSyncStorageTrie(db ethdb.KeyValueWriter, root common.Hash, account common.Hash) error {
+	bytes := make([]byte, 0, syncStorageTriesKeyLength)
+	bytes = append(bytes, syncStorageTriesPrefix...)
+	bytes = append(bytes, root[:]...)
+	bytes = append(bytes, account[:]...)
+	return db.Put(bytes, []byte{0x01})
+}
+
+// ClearSyncStorageTrie removes all storage trie accounts (with the given root) from db.
+// Intended for use when the trie with root has completed syncing.
+func ClearSyncStorageTrie(db ethdb.KeyValueStore, root common.Hash) error {
+	accountsPrefix := make([]byte, len(syncStorageTriesPrefix)+common.HashLength)
+	copy(accountsPrefix, syncStorageTriesPrefix)
+	copy(accountsPrefix[len(syncStorageTriesPrefix):], root[:])
+	return clearPrefix(db, accountsPrefix, syncStorageTriesKeyLength)
+}
+
+// ClearAllSyncStorageTries removes all storage tries added for syncing from db
+func ClearAllSyncStorageTries(db ethdb.KeyValueStore) error {
+	return clearPrefix(db, syncStorageTriesPrefix, syncStorageTriesKeyLength)
+}
+
+// ParseSyncStorageTrieKey returns the root and account for a storage trie
+// key returned from NewSyncStorageTriesIterator.
+func ParseSyncStorageTrieKey(keyBytes []byte) (common.Hash, common.Hash) {
+	keyBytes = keyBytes[len(syncStorageTriesPrefix):] // skip prefix
+	root := common.BytesToHash(keyBytes[:common.HashLength])
+	account := common.BytesToHash(keyBytes[common.HashLength:])
+	return root, account
+}
+
+// WriteSyncPerformed logs an entry in `db` indicating the VM state synced to `blockNumber`.
+func WriteSyncPerformed(db ethdb.KeyValueWriter, blockNumber uint64) error {
+	syncPerformedPrefixLen := len(syncPerformedPrefix)
+	bytes := make([]byte, syncPerformedKeyLength)
+	copy(bytes[:syncPerformedPrefixLen], syncPerformedPrefix)
+	binary.BigEndian.PutUint64(bytes[syncPerformedPrefixLen:], blockNumber)
+	return db.Put(bytes, []byte{0x01})
```

Comment:
Why do we write a `[]byte{0x01}` here?
```suggestion
	return db.Put(bytes, nil)
```